Description: the close and ctrl<w> buttons are now mapped to hide the main
             window when a music is playing (we can bring it back using the
             soundmenu). If no music is playing or if Music -> Quit is
             triggered rhythmbox is closed.
Author: Didier Roche
Bug-Ubuntu: https://bugs.launchpad.net/bugs/526552

Index: rhythmbox/shell/rb-shell.c
===================================================================
--- rhythmbox.orig/shell/rb-shell.c
+++ rhythmbox/shell/rb-shell.c
@@ -1950,11 +1950,42 @@ rb_shell_window_configure_cb (GtkWidget
 	return FALSE;
 }
 
+/**
+ * rb_shell_hide_and_not_quit:
+ * @shell: a RBShell
+ *
+ * Tell the application to hide instead of quitting if a song is currently
+ * playing.
+ *
+ * Return value: TRUE is hide (and stop quitting) or FALSE for quit
+ */
+gboolean
+rb_shell_hide_and_not_quit (RBShell *shell)
+{
+	gboolean playing;
+	GtkWindow *window;
+
+	if (rb_shell_player_get_playing (shell->priv->player_shell, &playing, NULL)
+		&& playing) {
+
+		g_object_get (shell, "window", &window, NULL);
+		gtk_widget_hide (window);
+		g_object_unref (window);
+
+		return TRUE;
+	}
+	return FALSE;
+}
+
 static gboolean
 rb_shell_window_delete_cb (GtkWidget *win,
 			   GdkEventAny *event,
 			   RBShell *shell)
 {
+
+	if (rb_shell_hide_and_not_quit (shell))
+		return TRUE;
+
 	if (shell->priv->party_mode) {
 		return TRUE;
 	}
Index: rhythmbox/shell/rb-shell.h
===================================================================
--- rhythmbox.orig/shell/rb-shell.h
+++ rhythmbox/shell/rb-shell.h
@@ -158,6 +158,8 @@ RBSource * rb_shell_get_source_by_entry_
 
 gboolean        rb_shell_get_party_mode (RBShell *shell);
 
+gboolean        rb_shell_hide_and_not_quit (RBShell *shell);
+
 void 		rb_shell_append_display_page (RBShell *shell, RBDisplayPage *page, RBDisplayPage *parent);
 
 void 		rb_shell_add_widget (RBShell *shell, GtkWidget *widget, RBShellUILocation location, gboolean expand, gboolean fill);
